Person overview mask badge (EXPOSUREAPP-13803)

schauersbergern opened the pull request:
Add Mask Badge in different states

Current CCL configuration leads to following UI states:

**MASK_OPTIONAL no G Status**
Scanned following certificate:
<img width="433" alt="MicrosoftTeams-image" src="https://user-images.githubusercontent.com/22813752/186440947-99f990f8-32b6-47d3-b7e5-a1cb80ad2709.png">

![Screenshot_20220824-154919](https://user-images.githubusercontent.com/22813752/186439471-5fb9cdc5-3341-47a7-82c8-092cbb1b3af9.png)
![Screenshot_20220824-154926](https://user-images.githubusercontent.com/22813752/186439483-778e93be-3bda-4ba1-af07-acef1a6d4b57.png)

**MASK_REQUIRED no G Status**
`cwa vc gen -e mock-cloud --preset biontech12`
![Screenshot_20220824-155546](https://user-images.githubusercontent.com/22813752/186439751-12b966ce-2451-404a-a7d5-4b341d740a7b.png)
![Screenshot_20220824-155552](https://user-images.githubusercontent.com/22813752/186439771-b74db8b8-f23b-4a48-b303-e582cac9149a.png)

**No Mask Info,  3G+ Status**
`cwa tc gen -e mock-cloud`
![Screenshot_20220824-155605](https://user-images.githubusercontent.com/22813752/186440498-6327adee-4433-44fb-993f-ad652d50d647.png)
![Screenshot_20220824-155613](https://user-images.githubusercontent.com/22813752/186440519-a501e0cd-a5ee-4f33-8221-44c989bbda14.png)

**No Mask Info, no G status, invalid certificate**
`cwa vc gen -e mock-cloud --preset biontech22`
![Screenshot_20220824-155816](https://user-images.githubusercontent.com/22813752/186440682-8308d5b0-a1af-4f68-a753-fe3e1f280d0f.png)

schauersbergern committed:
add mask badge

Ein-Tim commented:
Will "Masken Pflicht" (with a whitespace) be the final version of the term? I rather suggest to use "Maskenpflicht", like it's done e.g. here https://www.bundesregierung.de/breg-de/themen/coronavirus/corona-bundeslaender-1745198.

mtwalli commented:
> Will "Masken Pflicht" (with a whitespace) be the final version of the term? I rather suggest to use "Maskenpflicht", like it's done e.g. here https://www.bundesregierung.de/breg-de/themen/coronavirus/corona-bundeslaender-1745198.

@Ein-Tim the text you see is not final yet. it is also coming from CCL which means it can change at any time

mtwalli commented:
**1- Please update screenshots tests** 
2- Badges are not aligned together and they are  squarish not rounded as per design ,see :
<img width="460" alt="Screenshot 2022-08-25 at 09 25 08" src="https://user-images.githubusercontent.com/25054729/186603370-38344e8d-9f6d-4f55-b1dd-1c1077af1fef.png">

3- Bigger text size is not nice and the tile `Digitales...` is not completely visible 
<img width="435" alt="Screenshot 2022-08-25 at 09 29 06" src="https://user-images.githubusercontent.com/25054729/186603745-af11cfc7-a1f2-495c-a847-ac762cbe05ec.png">
<img width="452" alt="Screenshot 2022-08-25 at 09 29 49" src="https://user-images.githubusercontent.com/25054729/186603758-31946edb-d7f4-4945-959b-646f4398f4f0.png">

4- Sometimes I see white text when it should not see the video:
https://user-images.githubusercontent.com/25054729/186604010-5a914ccb-961a-4abf-b590-590b31fc30e4.mp4

schauersbergern committed:
add screenshot tests

schauersbergern committed:
adapt design

schauersbergern committed:
add test for invalid certificate

schauersbergern committed:
change status badge font size

schauersbergern committed:
Text adjustments

schauersbergern committed:
Text adjustments

schauersbergern commented:
> **1- Please update screenshots tests** 2- Badges are not aligned together and they are squarish not rounded as per design ,see : <img alt="Screenshot 2022-08-25 at 09 25 08" width="460" src="https://user-images.githubusercontent.com/25054729/186603370-38344e8d-9f6d-4f55-b1dd-1c1077af1fef.png">
> 
> 3- Bigger text size is not nice and the tile `Digitales...` is not completely visible <img alt="Screenshot 2022-08-25 at 09 29 06" width="435" src="https://user-images.githubusercontent.com/25054729/186603745-af11cfc7-a1f2-495c-a847-ac762cbe05ec.png"> <img alt="Screenshot 2022-08-25 at 09 29 49" width="452" src="https://user-images.githubusercontent.com/25054729/186603758-31946edb-d7f4-4945-959b-646f4398f4f0.png">
> 
> 4- Sometimes I see white text when it should not see the video:
> 
>  mask2-2.27.mp4 
> **### Please follow design specs 🤓 :** <img alt="Screenshot 2022-08-25 at 09 39 42" width="530" src="https://user-images.githubusercontent.com/25054729/186605007-973168cc-39e1-46f7-be62-d094989f7b35.png">

1) Screenshot Test Update done
2) Badges aligned with min size and same font size, corner radius changed
3) Adjusted certificate subtitle behaviour
4) I can't reproduce this atm. I think it was a dcc issue where the text was missing. Could you please recheck?

mtwalli committed:
Merge branch 'feature/13746-mark-requirements' into feature/13803_person_overview_mask_badge

schauersbergern committed:
Add maskBadgeTextColor
Fix padding

schauersbergern committed:
Add maskBadgeTextColor
Fix padding

schauersbergern committed:
Fix detekt

schauersbergern committed:
Fix test

schauersbergern committed:
Adapt badges visibility
Add mask badge line break height layout for both badges
Unify style for status badge on overview and vaccination detail

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt:
```diff
@@ -137,10 +137,22 @@ fun PersonOverviewItemBinding.setUIState(
+        }
+        statusBadge.setBackgroundResource(color.admissionBadgeBg)
+        statusBadge.text = statusBadgeText
+        if (statusBadgeText.isEmpty() && !item.hasMaskState) {
+            statusBadge.visibility = View.GONE
+        } else if (statusBadgeText.isEmpty() && item.hasMaskState) {
+            statusBadge.visibility = View.INVISIBLE
+        } else if (statusBadgeText.isNotEmpty()) {
+            statusBadge.visibility = View.VISIBLE
+        }
```
looks like a good candidate of `when` instead

schauersbergern replied:
Changed to when

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/CertificateStateHelper.kt:
```diff
@@ -137,10 +137,22 @@ fun PersonOverviewItemBinding.setUIState(
+        if (item.maskBadgeText.isEmpty()) {
+            maskBadge.visibility = View.INVISIBLE
+        } else {
+            maskBadge.visibility = View.VISIBLE
```
`maskBadge.isInvisible = item.maskBadgeText.isEmpty()` ?

schauersbergern replied:
simplified statement

mtwalli commented on Corona-Warn-App/src/main/res/values-night/colors.xml:
```diff
@@ -93,5 +93,7 @@
     <color name="maskGStatusBadgeColor">#31A056</color>
     <color name="maskBadgeColor">#FFFFFF</color>
+    <color name="maskBadgeTextColorRegular">#FFFFFF</color>
+    <color name="maskBadgeTextColorGreenBg">#FFFFFF</color>
```
no need to add it to dark file since it is the same as light (default )maskBadgeTextColorGreenBg

schauersbergern replied:
Removed redundant color

schauersbergern committed:
Remove redundant night color
Simplify statements

mtwalli committed:
Merge branch 'feature/13746-mark-requirements' into feature/13803_person_overview_mask_badge

chiljamgossow reviewed (commented):
Tested, looks good.

The space between the mask badge and the right edge when there is no g status looks different from the design. I suggest adding a minWidth to the g status badge of 40dp to match the design, or discuss making it full width in case there is no g status

chiljamgossow commented on Corona-Warn-App/src/main/res/values/styles.xml:
```diff
@@ -683,4 +683,13 @@
+        <item name="android:paddingEnd">6dp</item>
+        <item name="android:minHeight">26dp</item>
+        <item name="android:gravity">center</item>
+        <item name="android:textColor">#FFFFFF</item>
```
here you can add the minWidth of 40dp

mtwalli replied:
@schauersbergern  can you pls have a look ?

schauersbergern replied:
Which minWidth? In the design there is just a minHeight of 26dp
https://www.figma.com/file/KuDEcMSREg3VhZRhC3FXTY/COVID19_Master_2.25?node-id=55153%3A40951

schauersbergern committed:
Add min width
